tools/packaging/build/agent-enclave-bundle/Dockerfile

```
@@ -1,5 +1,9 @@
+ARG RATS_TLS_REV=5de6fc308f2a18f4105429451c85e6173b42ab9d
+
 FROM ubuntu:20.04 as builder
 
+ARG RATS_TLS_REV
```

maybe just define once:
```suggestion
ARG RATS_TLS_REV=5de6fc308f2a18f4105429451c85e6173b42ab9d
```

---

The `RATS_TLS_REV`  is designed for the two-stage building. The `RATS_TLS_REV` in defined and initialized in line 1. And then it can be used directly in the builder and the real container by re-declaring it again. I think the design can be reserved.

---

what is the use-case where it is needed outside the builder stage?

---

For example, the agent depends on the dynamic lib `rats-tls`, the lib need to be installed in the builder to complete compiling, and in the real container to enable the agent running upon the bundle. 

And It is duplicated to define and initialize to two `RATS_TLS_REV` twice in the builder and the real container respectively, which looks like:
```
From ... as builder
ARG RATS_TLS_REV=5de6fc308f2a18f4105429451c85e6173b42ab9d

From ...
ARG RATS_TLS_REV=5de6fc308f2a18f4105429451c85e6173b42ab9d
```
It is also easy for us to forget to  update these two `RATS_TLS_REV` at the same time.

But our design will looks like:
```
ARG RATS_TLS_REV=5de6fc308f2a18f4105429451c85e6173b42ab9d
From ... as builder
ARG RATS_TLS_REV

From ...
ARG RATS_TLS_REV
```
It remedies the downside of the former design.

---

> And It is duplicated to define and initialize to two `RATS_TLS_REV` twice in the builder and the real container

AFAICS it's only needed once (in the builder).

---

But as I know, the lib `rats-tls` installed in the builder will not appear in the real container, when the agent is launched, the `ld` can not find the `rats-tls`.

---

That's not correct. It needs to be installed in the Occlum instance only and the Occlum package (after it's built) containers `rats-tls`  and can be copied anywhere.

---

If the builder is responsible for compiling the agent and packaging the occlum instance, the `rats-tls` can be installed on once.